include/ttmlir/Dialect/TTIR/IR/TTIROps.td

```diff
@@ -181,6 +181,13 @@ def TTIR_RsqrtOp : TTIR_ElementwiseUnaryOp<"rsqrt"> {
     }];
 }
 
+def TTIR_RealOp : TTIR_ElementwiseUnaryOp<"real"> {
+    let summary = "Eltwise real.";
+    let description = [{
+      Eltwise real operation.
+    }];
+}
```

Comment:
It's kinda mystic to me what this op does. 
1. Please enhance description,
2. I am not sure why are we adding this op without proper backend support, what is the use case for it?

Reply:
Seems like we need this OP to support the LLaMA model, LLaMA uses [rotary embeddings](https://arxiv.org/pdf/2104.09864v5) which are based on complex number arithmetics.

As Radenko said, we don't have much use of supporting this in TTIR only, without proper backend support. We should look into Metal ttnn API support for complex number arithmetics and see how to bring it up in TTMLIR. I can take a look when I find some time but also FYI for @mmanzoorTT who is involved in LLaMA bringup.

Reply:
I agree that we need proper backend support. We want to run LLaMA inference through `tt-xla` which will generate graph in `stablehlo` dialect.

Reply:
It seems the purpose of this real Op is to extract the _`real`_ portion of the complex tensor, can you create a ttnn op that just use this function to extract this?

```
Tensor _real(const ComplexTensor& input, const MemoryConfig& output_mem_config) {
    return input[0];
}
```

Reply:
I'm wondering if we should just lower `complex` to be `2xf32` and then `stablehlo.real` just becomes `ttir.slice(dim=-1, index=0)` and `stable.complex` becomes `ttir.slice(dim=-1, index=1)`.  `slice` op should actually copy `ttnn/cpp/ttnn/operations/data_movement/slice/slice.hpp`, just wanted to illustrate the point.

So concrete example, we have input IR:
```mlir
%result = "stablehlo.real"(%operand) : (tensor<2xcomplex<f32>>) -> tensor<2xf32>
```

After type conversion:
```mlir
%result = "stablehlo.real"(%operand) : (tensor<2x2xf32>) -> tensor<2xf32>
```

After op conversion:
```mlir
%result = "ttir.slice"(%operand, [:, 0]) : (tensor<2x2xf32>) -> tensor<2xf32>
```

Reply:
Here is my understand, please correct me if I'm wrong.

1) We need to implement stablehlo.complex conversion. This can then be used with ComplexTensor type directly. 
2) We need to use ttir.slice to implement real and imag

If this sounds good, i could start working on complex instead and close this PR for the time being.

Reply:
@ddilbazTT that seems like a good approach.

Reply:
Hi everyone, I started working on ComplexOp and hit a roadblock on how to interchange between `Tensor/ComplexTensor` structs. The discussion is [here](https://github.com/tenstorrent/tt-mlir/issues/748) I would appreciate if you could give any tips on the ComplexOp implementation.